crypto/asn1/d2i_pr.c

```diff
@@ -58,6 +58,8 @@ EVP_PKEY *d2i_PrivateKey_ex(int type, EVP_PKEY **a, const unsigned char **pp,
                 goto err;
             EVP_PKEY_free(ret);
             ret = tmp;
+            if (EVP_PKEY_type(type) != EVP_PKEY_base_id(ret))
```

Comment:
This change deserve a line of comment, IMHO

Reply:
This is the fix @levitte provided above; it's in an extra commit with the message
```
Fix d2i_PrivateKey{,_ex} to check for the expected type
```